test/Microsoft.AspNet.Mvc.Core.Test/ApplicationModel/DefaultActionModelBuilderTest.cs

```diff
@@ -646,6 +646,22 @@ public void GetActions_MixedHttpVerbsAndRoutes_MultipleEmptyAndNonEmptyVerbs()
             Assert.Equal<string>(new string[] { "GET" }, action.HttpMethods);
         }
 
+        [Fact]
```

Comment:
In both cases it will throw.. I thought about it but did not seem worth fixing that, derived models with different way of binding sounds weird.Same thing goes for types as well. The workaround would be to have a different model. do you think otherwise? ( In any case I will be adding tests for the second one).